client/src/main/java/org/apache/rocketmq/client/producer/MQProducer.java

```diff
@@ -91,6 +91,10 @@ TransactionSendResult sendMessageInTransaction(final Message msg,
     SendResult send(final Collection<Message> msgs) throws MQClientException, RemotingException, MQBrokerException,
         InterruptedException;
 
+    // for batch msgs with multi topics.
+    SendResult sendMultiTopicBatch(final Collection<Message> msgs) throws MQClientException, RemotingException, MQBrokerException,
+            InterruptedException;
+
```

Comment:
Who has the responsibility to make sure the messages are to be sent to the same broker? How to make sure it?

This API may cause many difficulties to be compatible with advanced features, such as logic queue and message with sharding key?

Only the inner topic route has enough information to decide which queue to be sent for each message. So the multi-topic batch can only be constructed by the inner MQClientInstance, but not by the user directly.

IMO, it is necessary to raise a RIP and provide a detailed design for this feature.